ldk/javascript/src/whisper/index.ts

```diff
@@ -265,6 +284,146 @@ export interface WhisperAptitude {
+    case WhisperComponentType.ListPair:
+    case WhisperComponentType.Markdown:
+    case WhisperComponentType.Message:
+      return component;
+    case WhisperComponentType.Number:
+      return {
+        ...component,
+        onChange: (error, param, whisper) => {
+          component.onChange(error, param, convertGojaWhisper(whisper));
+        },
+      } as OliveHelps.NumberInput;
+    case WhisperComponentType.Password:
+      return {
+        ...component,
+        onChange: (error, param, whisper) => {
+          component.onChange(error, param, convertGojaWhisper(whisper));
+        },
+      } as OliveHelps.Password;
+    case WhisperComponentType.RadioGroup:
+      return {
+        ...component,
+        onSelect: (error, param, whisper) => {
+          component.onSelect(error, param, convertGojaWhisper(whisper));
+        },
+      } as OliveHelps.RadioGroup;
+    case WhisperComponentType.Select:
+      return {
+        ...component,
+        onSelect: (error, param, whisper) => {
+          component.onSelect(error, param, convertGojaWhisper(whisper));
+        },
+      } as OliveHelps.Select;
+    case WhisperComponentType.Telephone:
+      return {
+        ...component,
+        onChange: (error, param, whisper) => {
+          component.onChange(error, param, convertGojaWhisper(whisper));
+        },
+      } as OliveHelps.Telephone;
+    case WhisperComponentType.TextInput:
+      return {
+        ...component,
+        onChange: (error, param, whisper) => {
+          component.onChange(error, param, convertGojaWhisper(whisper));
+        },
+      } as OliveHelps.TextInput;
+    default:
+      throw new Error('Unexpected component type');
+  }
+}
+
+function convertComponent(component: Components): OliveHelps.Components {
+  switch (component.type) {
+    case WhisperComponentType.CollapseBox:
+      return {
+        label: component.label,
+        open: component.open,
+        children: component.children.map(convertChildComponent),
+        type: WhisperComponentType.CollapseBox,
+      };
+    default:
+      return convertChildComponent(component);
+  }
+}
+
+function convertToGojaWhisper(whisper: UpdateWhisper): OliveHelps.UpdateWhisper;
+function convertToGojaWhisper(whisper: NewWhisper): OliveHelps.NewWhisper;
+function convertToGojaWhisper(
+  whisper: NewWhisper | UpdateWhisper,
+): OliveHelps.NewWhisper | OliveHelps.UpdateWhisper {
+  return 'onClose' in whisper
+    ? {
+        label: whisper.label,
+        onClose: whisper.onClose,
+        components: whisper.components.map(convertComponent),
+      }
+    : {
+        label: whisper.label,
+        components: whisper.components.map(convertComponent),
+      };
+}
+
+function convertGojaWhisper(whisper: OliveHelps.Whisper): Whisper {
+  return {
+    id: whisper.id,
+    close: whisper.close,
+    update(updateWhisper: UpdateWhisper, cb): void {
+      whisper.update(convertToGojaWhisper(updateWhisper), cb);
```

Comment:
It's very important that `whisper.update()` is called on a converted whispers. Can a test be added for this?

Reply:
Sure!

Reply:
Added!